packages/back-end/src/routers/organizations/organizations.controller.ts

```diff
@@ -1002,14 +1012,35 @@ export async function signup(req: AuthRequest<SignupBody>, res: Response) {
     }
   }
 
+  let verifiedDomain = "";
+  if (IS_CLOUD) {
+    // if the owner is verified, try to infer a verified domain
+    if (req.userId) {
+      const user = await findUserById(req.userId);
+      if (user?.verified) {
+        const domain = user.email.toLowerCase().split("@")[1] || "";
+        const isFreeDomain = freeEmailDomains.includes(domain);
+        if (!isFreeDomain) {
+          verifiedDomain = domain;
+        }
+      }
+    }
+  }
+
   try {
     if (company.length < 3) {
       throw Error("Company length must be at least 3 characters");
     }
     if (!req.userId) {
       throw Error("Must be logged in");
     }
-    const org = await createOrganization(req.email, req.userId, company, "");
+    const org = await createOrganization(
+      req.email,
+      req.userId,
+      company,
+      "",
+      verifiedDomain
+    );
```

Comment:
We should probably change the signature of this method to use an options object to avoid human error with the last 3 being strings. It would also read better since it's not clear what the empty string is.

Reply:
sounds good